doc/nrf/ug_contribution.rst

```diff
@@ -0,0 +1,127 @@
+rebasing easier, and keeps the change history clear and clean.
+
+When contributing to NCS, it is also important you provide as much
+information as you can about your change, update appropriate documentation,
+and test your changes thoroughly before submitting.
+
+The general GitHub workflow used by NCS developers uses a combination of
+command line Git commands and browser interaction with GitHub.  As it is with
+Git, there are multiple ways of getting a task done.  We'll describe a typical
+workflow here:
+
+.. _Create a Fork of NCS manifest repo (fw-nrfconnect-nrf):
+  https://github.com/NordicPlayground/fw-nrfconnect-nrf#fork-destination-box
+
+#. `Create a Fork of fw-nrfconnect-nrf`_
+   to your personal account on GitHub. (Click on the fork button in the top
+   right corner of the fw-nrfconnect-nrf repo page in GitHub.)
+
+#. On your development computer, change into the :file:`nrf` folder that was
+   created when you :ref:`obtained the code <get_the_code>`::
+
+     cd ncs/nrf
+
+   Rename the default remote pointing to the `upstream repository
+   <https://github.com/NordicPlayground/fw-nrfconnect-nrf>`_ from ``origin`` to
+   ``upstream``::
+
+     git remote rename origin upstream
+
+   Let Git know about the fork you just created, naming it ``origin``::
+
+     git remote add origin https://github.com/<your github id>/fw-nrfconnect-nrf
+
+   and verify the remote repos::
+
+     git remote -v
+
+   The output should look similar to::
+
+     origin   https://github.com/<your github id>/fw-nrfconnect-nrf (fetch)
+     origin   https://github.com/<your github id>/fw-nrfconnect-nrf (push)
+     upstream https://github.com/NordicPlayground/fw-nrfconnect-nrf (fetch)
+     upstream https://github.com/NordicPlayground/fw-nrfconnect-nrf (push)
+
+#. Create a topic branch (off of master) for your work (if you're addressing
+   an issue, we suggest including the issue number in the branch name)::
+
+     git checkout master
+     git checkout -b fix_comment_typo
+
+
+#. Make changes, test locally, change, test, test again, ...  (Check out the
+   prior chapter on `sanitycheck`_ as well).
+
+#. When things look good, start the pull request process by adding your changed
+   files::
+
+     git add [file(s) that changed, add -p if you want to be more specific]
+
+   You can see files that are not yet staged using::
+
+     git status
+
+#. Verify changes to be committed look as you expected::
+
+     git diff --cached
+
+#. Commit your changes to your local repo::
+
+     git commit -s
+
+   The ``-s`` option automatically adds your ``Signed-off-by:`` to your commit
+   message.  Your commit will be rejected without this line that indicates your
+   agreement with the `DCO`_.  See the `Commit Guidelines`_ section for
+   specific guidelines for writing your commit messages.
+
+#. Push your topic branch with your changes to your fork in your personal
+   GitHub account::
+
+     git push origin fix_comment_typo
```

Comment:
This should be:
`git push -u origin fix_comment_typo`
so that the branch is tracked after